### ISSUE TITLE
How can I get Detection Sequence?

### ISSUE THREAD

**Issue by Wangzhen-kris** [aimagelab/show-control-and-tell#4]
Hi, thanks for your great work!
      I have a question and need your help. How can I get the Detection Sequence for all regions in each image? And I want to know how to guarantee the Detection Sequence of the same shape, because the number of regions in each image is not the same.
      What's more, I found this sentence in your code:  `_, _, ctrl_det_seqs_test, _, captions = values`. So `ctrl_det_seqs_test` is the Detection Sequence for all regions features in each image?
      If so, can the Detection Sequence of the training data be obtained in the same way?
      In short, I want to get the Detection Sequence for each image of coco data.
      Can you help me? Thanks a lot!!

**Comment by JonasWangCN** [comment]
Hello？ And then I want to know how many boxes were sorted?

**Comment by marcellacornia** [comment]
Hi @JonasWangCN,
thanks for downloading our code and sorry for the late reply.

The current code is for the testing phase, only. However, you can use the current dataloader to get the detection sequence for each image-caption pair. `ctrl_det_seqs_test` contains the detection sequence for an image-caption pair padded accordingly when needed.

Despite some delays, we plan to release the training code soon. I hope it can be useful!

**Comment by marcellacornia** [comment]
Hi @JonasWangCN,
the training code is now available. 

`ctrl_det_seqs` contains the detection sequence for a batch of COCO images (line 105 of the `train.py` file).
